browse/services/database/listings.py

```diff
@@ -347,10 +347,11 @@ def get_articles_for_month(
                 | (doc.paper_id.like(f"%/{year % 100:02d}%"))
             )                     
   
+    doc_ids_sub = doc_ids.subquery()
     cat_conditions = [and_(aic.c.archive == arch_part, aic.c.subject_class == subj_part) for arch_part, subj_part in cats]
     #filters to only the ones in the right category and records if any of the requested categories are primary
     cat_query = (session.query(aic.c.document_id, func.max(aic.c.is_primary).label('is_primary'))
-        .where(aic.c.document_id.in_(doc_ids))
+        .join(doc_ids_sub, doc_ids_sub.c.document_id == aic.c.document_id)
```

Comment:
Do we want to make these even minor changes? Or do we just want to time what we have?

Reply:
It passes tests, it's pretty simple logically, and I've tested it pretty extensively in cloud run. It's also just more correct